### ISSUE TITLE
Hyprwinwrap works randomly

### ISSUE THREAD

**Issue by offeex** [hyprwm/hyprland-plugins#100]
Look, I don't mean any disrespect to the main developer behind hyprwinwrap, but holy fucking shit has this been an absolute nightmare to setup. I've been bashing my head against the wall trying to get a simple looping video wallpaper working with mpv.

I launch my system, thinking "Alright, easy peasy, just `mpv video.mp4 --loop` in the terminal." But oh no, that's just the start of my fucking Odyssey with this contraption. Nothing happens. No video, no wallpaper, just empty void laughing in my face.
So I start going through all the troubleshooting steps like a goddamn tech priest. Enabling, disabling, reloading this goddamn plugin manager - you name it, I've tried it fifty fucking times over. Didn't make a lick of difference.

The only way I can sometimes, and I mean SOMETIMES, get this bastard child of Cthulhu and a dumpster fire to even maybe work, is by opening the eye-gouging **config file**. Then I have to furiously switch between `class=mpv` and `class = mpv` like I'm a circus bear being poked with a hot iron.
Even on the rare miracle occasions when it halfway works, I feel like I've been mentally waterboarded just to get a fucking video loop as my wallpaper. I'm talking 'Nam flashback levels of trauma just from this shit.

Look, I get that software development is hard, but god damn did they fuck the dog on this one. If I were a clinic-level masochist, maybe I'd get off on this amount of struggle-fucking just to accomplish a basic task. But alas, I'm just a normal guy who wanted a little video wallpaper bling.
I'm this close to just saying "fuck it" and becoming a Trappist monk to escape this unholy hell-realm of suffering. Or maybe I'll take up birdwatching. Either way, it'll be more peaceful than dealing with this swamp fuckery ever again.

**Comment by offeex** [comment]
> uhhh... https://github.com/GhostNaN/mpvpaper

well, does not work on nvidia gpu unfortunately. Only iGPU. mpv works flawlessly tho

**Comment by vaxerski** [comment]
that's odd. Anyways, feel free to debug yourself too, it's a small plugin :)

**Comment by atikiNBTW** [comment]
yeah, the plugin will only work if it sees update in the hyprland.conf. If you start the mpv in exec-once, you will need to make a script to copy the hyprland.conf to itself to get hyprwinwrap to work

**Comment by Teenarp2003** [comment]
I have been having the same problem . It really works intermittently. has anybody found a fix for this? i have trawled through the entire internet to get a fix. however i tried it does not work reliably for GLava. as @offeex commented flipping though the spaces between the class = GLava and class=GLava worked but this is so intermittent and every time i relogin it stops working.

**Comment by atikiNBTW** [comment]
Hyprland actually does not automatically reload the config after plugin init. If you add `g_pConfigManager->reload();` after registering the config value it works normally. I think this is a problem with Hyprland, but I may be doing something wrong. 
![image](https://github.com/user-attachments/assets/20ff8fbb-355a-43a6-a071-3b68ee84f04e) <- not true, proof:


https://github.com/user-attachments/assets/26efbc1f-37ce-4798-b594-f54fd203a759
